docs/build/voting-dapp-tutorial.md

```diff
@@ -0,0 +1,1155 @@
+:::
+
+### Gas Station
+
+A unique feature of Kadena is the ability to allow gas to be paid by a different entity than the one who initiated the transaction. This entity is what we call a *gas station*.
+
+ In our voting app this will allow users to submit votes without paying for gas, instead gas will be subsidized by the gas station.
+
+The standard for gas station implementation is defined by the `gas-payer-v1` interface. The `gas-payer-v1` interface is deployed to all chains on `testnet` and `mainnet` so you can directly use it in your contract. We can specify that a module implements an interface using the `(implements INTERFACE)` construct.
+
+:::info
+
+Pact interfaces are similar to Java's interfaces, Scala's traits, Haskell's typeclasses or Solidity's interfaces.
+If you're not familiar with this concept you can read more about it <a href="https://pact-language.readthedocs.io/en/latest/pact-reference.html#interfaces">**here**</a>.
+
+:::
+
+```clojure
+(interface gas-payer-v1
+
+  (defcap GAS_PAYER:bool
+    ( user:string
+      limit:integer
+      price:decimal
+    )
+    @doc
+    " Provide a capability indicating that declaring module supports \
+    \ gas payment for USER for gas LIMIT and PRICE. Functionality \
+    \ should require capability (coin.FUND_TX), and should validate \
+    \ the spend of (limit * price), possibly updating some database \
+    \ entry. \
+    \ Should compose capability required for 'create-gas-payer-guard'."
+    @model
+    [ (property (user != ""))
+      (property (limit > 0))
+      (property (price > 0.0))
+    ]
+  )
+
+  (defun create-gas-payer-guard:guard ()
+    @doc
+    " Provide a guard suitable for controlling a coin account that can \
+    \ pay gas via GAS_PAYER mechanics. Generally this is accomplished \
+    \ by having GAS_PAYER compose an unparameterized, unmanaged capability \
+    \ that is required in this guard. Thus, if coin contract is able to \
+    \ successfully acquire GAS_PAYER, the composed 'anonymous' cap required \
+    \ here will be in scope, and gas buy will succeed."
+  )
+)
+```
+
+Our module needs to implement all the functions and capabilities defined by the `gas-payer-v1` interface:
+* `GAS_PAYER` capability
+* `create-gas-payer-guard` function
+
+A gas station allows someone to debit from a coin account that they do not own, gas station account, to pay the gas fee for a transaction under certain conditions. How exactly that happens, let's see below.
+
+Create a new file `election-gas-station.pact` and paste the following snippet:
+
+```clojure
+(module election-gas-station GOVERNANCE
+  (defcap GOVERNANCE ()
+    "Only admin account can update the smart contract"
+    (enforce-keyset 'election-admin-keyset))
+
+  ; Implement the gas-payer-v1 interface
+  (implements gas-payer-v1)
+
+  ; Import the coin module
+  (use coin)
+)
+```
+
+Now let's implement the `gas-payer-v1` interface:
```

Comment:
This code snippet below definitely needs more explanation. Where is `chain-data` coming from? What is it? What does it mean to be an `exec` transaction? Why do we test that the length is 1 at `exec-code`? Where did the `free-election.` module come from? Why are we taking 15 from the exec-code? What does it mean to compose capabilities (ie. the call to `compose-capability`)?

I feel like a lot of code in this snippet below requires introduction and an explanation as to what it's doing and why.